File: hw/maisincho/u3/HW22Strategy/Strategy/src/main/java/ec/edu/espe/strategy/controller/SortingContext.java

package ec.edu.espe.strategy.controller;

import ec.edu.espe.strategy.model.BubbleSort;
import ec.edu.espe.strategy.model.InsertionSort;
import ec.edu.espe.strategy.model.QuickSort;
import ec.edu.espe.strategy.model.SortingStrategy;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public class SortingContext {
    private SortingStrategy strategy;
    private String strategyName;

    public void setSortStrategy(int[] array) {
        if(array.length<30){
        this.strategy = new BubbleSort();
        this.strategyName="Bubble Sort";
    }else if(array.length<100){
        this.strategy = new InsertionSort();
        this.strategyName="Insertion Sort";
    }else {
        this.strategy=new QuickSort();
        this.strategyName="Quick Sort";
    }
    
    }
    public void sortArray(int[] array){
        System.out.println("Original array: "+java.util.Arrays.toString(array));
        setSortStrategy(array);
        if(strategy !=null){
            strategy.sort(array);
            System.out.println("Sorted array:"+java.util.Arrays.toString(array));
            System.out.println("Sortin startegy used "+strategyName);
        }else{
            System.out.println("No sorting strategy set ");
        }
    }
}

File: hw/maisincho/u3/HW22Strategy/StrategyPy/controller/SortingContext.py
from typing import List
from model.BubbleSort import BubbleSort
from model.InsertionSort import InsertionSort
from model.QuickSort import QuickSort
from model.SortingStrategy import SortingStrategy

class SortingContext:
    def __init__(self):
        self.strategy: SortingStrategy = None
        self.strategy_name: str = ""

    def set_sort_strategy(self, array: List[int]) -> None:
        if len(array) < 30:
            self.strategy = BubbleSort()
            self.strategy_name = "Bubble Sort"
        elif len(array) < 100:
            self.strategy = InsertionSort()
            self.strategy_name = "Insertion Sort"
        else:
            self.strategy = QuickSort()
            self.strategy_name = "Quick Sort"

    def sort_array(self, array: List[int]) -> None:
        print("Original array:", array)
        self.set_sort_strategy(array)
        if self.strategy:
            self.strategy.sort(array)
            print("Sorted array:", array)
            print("Sorting strategy used:", self.strategy_name)
        else:
            print("No sorting strategy set.")
